### PR TITLE
fix(ui): Fix <ContextPickerModal> and refactor to ts

### PR DESCRIPTION
**Pull request by billyvg**
This removes and fixes `ContextPickerModal` and refactors it to ts.

Fixes JAVASCRIPT-153E

### CODE REVIEW

**Comment by billyvg** [comment]
cc @mitsuhiko

**Comment by matejminar** [comment]
Is this ready to be reviewed/merged?  I would like to resolve this related PR afterward https://github.com/getsentry/sentry/pull/16100

**Comment by billyvg** [comment]
Ah yeah sorry, need to fix/update tests.

File: src/sentry/static/sentry/app/components/contextPickerModal.tsx [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+          {needOrg && (
+            <StyledSelectControl
+              innerRef={ref => {
+                this.orgSelect = ref;
+                if (shouldShowProjectSelector) {
+                  return;
+                }
+                this.focusOrganizationSelector();
+              }}
+              placeholder="Select an Organization"
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/contextPickerModal.tsx]:
```suggestion
              placeholder={t('Select an Organization')}
```

File: src/sentry/static/sentry/app/components/contextPickerModal.tsx [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+            />
+          )}
+
+          {organization && needProject && projects && (
+            <StyledSelectControl
+              innerRef={ref => {
+                this.projectSelect = ref;
+                this.focusProjectSelector();
+              }}
+              placeholder="Select a Project"
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/contextPickerModal.tsx]:
```suggestion
              placeholder={t('Select a Project')}
```

**Review by matejminar — APPROVED** [review]

**Pull request merged**